core/amazon/index.js

```diff
@@ -150,7 +153,13 @@ var amazonProduct = {
                                 var variation = variationKeys[variationIdx];
                                 var value = itemAttributes[variation];
                                 if (!(value in ref)) {
+                                    // Otherwise, add value to map
                                     if (variationIdx == variationKeys.length - 1) {
+                                        // Check if the last level variations are too numerous for selection through conversation
+                                        if (Object.keys(ref).length == 10) {
+                                            conversational = false;
```

Comment:
So since I'm checking the length before pushing the next value then if the length is already 10 I know the next one will put it over... I don't mind moving the if statement after adding the value if that makes it more clear though